common/lib/common-utils/src/logger.ts

```diff
@@ -41,3 +41,41 @@ export class TelemetryNullLogger implements ITelemetryLogger {
     public sendPerformanceEvent(event: ITelemetryPerformanceEvent, error?: any): void {
     }
 }
+
+export function extractLogSafeErrorProperties(error: any) {
+    const isRegularObject = (value: any): boolean => {
+        return value !== null && !Array.isArray(value) && typeof value === "object";
+    };
+
+    const removeMessageFromStack = (stack: string, errorName?: string) => {
+        const stackFrames = stack.split("\n");
+        stackFrames.shift(); // Remove "[ErrorName]: [ErrorMessage]"
+        if (errorName !== undefined) {
+            stackFrames.unshift(errorName); // Add "[ErrorName]"
+        }
+        return stackFrames.join("\n");
+    };
```

Comment:
I think this is a style question more than anything "for my information".  These look like "pure" functions -- an reason not to hoist them to the top scope?

Reply:
Just feels more self-contained this way as I was moving it around 🤷‍♂️  Any reason not to leave them in here? I could also inline them but liked giving them a name.

Reply:
...mostly premature optimization that are likely of particular insignificance in this context (also my understanding may not even be correct in the "state of the art" of  common JS engines and/or minifying transpilers).  Basically every time this function is called two function objects (and space for binding their scopes) might be allocated.  Also (if it were of interest), the functions could be tested independently.

Alternatives when functions are only invoked once (without recursion) might include
```
// commented
orLabeled:
{
  // blocks
}
```
but this can start to err on the side of "style-weird".